File: server/src/main/java/com/github/mtgaccountant/server/models/Card.java
package com.github.mtgaccountant.server.models;

import java.io.Serializable;

import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

@Data
@Document("cards")
public class Card implements Serializable{
    private String id;
    
    private String name;

    private String released_at;

    private ImageUri image_uris;

    private String oracle_text;

    private String power;

    private String toughness;

    private String[] colors;

    private String[] color_identity;    

    private String[] keywords;

    private String set_name;

    private String set_type;

    private String collector_number;

    private String rarity;

    private String flavor_text;

    private String artist;

    private Prices prices;

    private PurchaseUri purchase_uris;

    private boolean collected;
}

File: client/src/app/models/Card.ts
import { CardFaces } from './CardFaces';
import { PurchaseUris } from './PurchaseUris';

export interface Card {
    id: string;
    name: string;
    released_at: string;
    image_uris: ImageUris;
    oracle_text: string;
    power: string;
    toughness: string;
    parent_set_code: string;
    icon_svg_uri: string;
    colors: string[];
    color_identity: string[];
    keywords: [];
    card_faces: CardFaces[];
    set: string;
    set_name: string;
    set_type: string;
    collector_number: string;
    rarity: string;
    flavor_text: string;
    artist: string;
    prices: Prices;
    purchase_uris: PurchaseUris;
    collected: boolean;
}

export interface ImageUris {
    small: string;
    normal: string;
    large: string;
    png: string;
    border_crop: string;
}

export interface Prices {
    usd: string;
    usd_foil: string;
    eur: string;
    eur_foil: string;
    tix: string;
}
